src/websocket/common.ts

```diff
@@ -1,7 +1,33 @@
 const ERROR_TYPE = 'error'
+const DEFAULT_TIMEOUT = 15000
 
 export const catchError = (res: any) => {
   return res.type === ERROR_TYPE
     ? Promise.reject(res)
     : Promise.resolve(res)
 }
+
+export const promiseTimeout = (promise: Promise<any>, ms?: number) => {
+  let id
+  let time = ms === undefined ? DEFAULT_TIMEOUT : ms
+
+  let timeout = new Promise((resolve, reject) => {
+    id = setTimeout(() => {
+      clearTimeout(id);
+      reject(new Error('Timed out in '+ time + 'ms.'))
```

Comment:
could use `tag` here (if it makes sense to do so)
```suggestion
      reject(new Error(`Timed out in ${time}ms: ${tag}'))
```